app/services/create_form_service.rb

```diff
@@ -1,7 +1,32 @@
 class CreateFormService
+  class CreateFormEvent < ApplicationRecord
+    belongs_to :user
+    belongs_to :group
+  end
+
   def create!(creator:, group:, name:)
-    form = FormRepository.create!(creator_id: creator.id, name:)
-    GroupForm.create!(group:, form_id: form.id)
+    event = begin
+      CreateFormEvent.create!(group:, form_name: name, user: creator, dedup_version: 1)
+    rescue ActiveRecord::RecordNotUnique
+      previous_event = CreateFormEvent.order(created_at: :desc).find_by(group:, form_name: name)
+      if previous_event.user == creator && previous_event.created_at > 1.second.ago
+        while previous_event.form_id.blank?
+          sleep(0.01)
```

Comment:
What if there is an error creating the form in forms-admin? I guess the request will just eventually time out? Will Rails kill the thread successfully?